seismicpro/gather/gather.py

```diff
@@ -1490,11 +1551,37 @@ def _finalize_plot(self, ax, title, divider, event_headers, top_header,
 
         top_ax.set_title(**{'label': None, **title})
 
+        if len(ax.get_legend_handles_labels()[0]):
+            ax.legend()
+
+    def _process_masks(self, masks):
+        colors_iterator = cycle(['tab:red', 'tab:blue', 'tab:orange', 'tab:green', 'tab:purple', 'tab:pink',
+                                 'tab:olive', 'tab:cyan'])
+        masks_list = self._parse_headers_kwargs(masks, "masks")
+        for ix, mask_dict in enumerate(masks_list):
+            mask = mask_dict["masks"]
+            if isinstance(mask, Gather):
+                mask = mask.data
+            elif isinstance(mask, str):
+                mask_dict["label"] = mask_dict.get("label", mask)
+                mask = self[mask]
+
+            mask = np.array(mask)
+            if mask.ndim == 1:
+                mask = mask.reshape(-1, 1)
+            threshold = mask_dict.pop("threshold", 0.5)
+            mask_dict["masks"] = np.broadcast_to(np.where(mask < threshold, np.nan, 1), self.shape)
+            mask_dict["label"] = mask_dict.get("label", f"mask_{ix+1}")
+            mask_dict["color"] = mask_dict.get("color", next(colors_iterator))
+            if np.isnan(mask_dict["masks"]).all():
+                continue
+            yield mask_dict
```

Comment:
```suggestion
        return [mask_dict for mask_dict in masks_list if not np.isnan(mask_dict["masks"]).all()]
```
Not sure if I'd like this method to be a generator